lib/shifts.rb

```diff
@@ -2,65 +2,49 @@
 
 class Shifts
 
-    def initialize
-      @shifts = shifts
-      @keys = keys
-      @offsets = offsets
-    end
-
     def random_number
       rand(99999)
     end
 
-    def padded_random_number
-      sprintf("%05d", random_number)
+    def padded_random_number(number = random_number)
+      sprintf("%05d", number)
     end
 
-    def random_number_array
-      padded_random_number.split(//).to_a
+    def random_number_array(number = random_number)
+      padded_random_number(number).split(//).to_a
     end
 
-    def key_pairs_array
+    def key_pairs_array(number = random_number)
       pairs_array = []
-      random_number_array.each_cons(2) do |set|
+      random_number_array(number.to_i).each_cons(2) do |set|
         pairs_array << set
       end
       pairs_array
     end
 
-    def keys
-      {
-        "A key" => key_pairs_array[0].join.to_i,
-        "B key" => key_pairs_array[1].join.to_i,
-        "C key" => key_pairs_array[2].join.to_i,
-        "D key" => key_pairs_array[3].join.to_i
-        }
+    def keys(number = random_number)
+      key_pairs_array(number).flat_map do |pair|
+        pair.join.to_i
+      end
     end
 
     def todays_date
       Date.today.strftime("%d%m%y")
     end
 
-    def square_date
-      todays_date.to_i * todays_date.to_i
+    def square_date(date = todays_date)
+      date.to_i * date.to_i
     end
 
-    def offsets
-      square_date_array = square_date.to_s.split(//).to_a
-      {
-        "A offset" => square_date_array[-4].to_i,
-        "B offset" => square_date_array[-3].to_i,
-        "C offset" => square_date_array[-2].to_i,
-        "D offset" => square_date_array[-1].to_i
-        }
+    def offsets(date = todays_date)
```

Comment:
Completley changed logic here too to create an array of offsets. A little trickier on this one to return integer values not strings.